entproto/cmd/protoc-gen-entgrpc/converter.go

```diff
@@ -127,32 +129,62 @@ func convertPbMessageType(md *desc.MessageDescriptor, entFieldType string, conv
 	return nil
 }
 
-func (g *serviceGenerator) renderToProto(conv *converter, ident string) string {
+func (g *serviceGenerator) renderToProto(conv *converter, varName, ident string) string {
 	var left, right string
-	if conv.toProtoConstructor.GoName != "" {
-		left += g.QualifiedGoIdent(conv.toProtoConstructor) + "("
+
+	if conv.toProtoTryConstructor.GoName != "" {
+		left += varName + ", err := " + g.QualifiedGoIdent(conv.toProtoTryConstructor) + "("
+		right += `)
+		if err != nil {
+			return nil, err
+		}
+		`
+	} else if conv.toProtoConstructor.GoName != "" {
+		left += varName + " := " + g.QualifiedGoIdent(conv.toProtoConstructor) + "("
 		right += ")"
 	}
 	if conv.toProtoConversion != "" {
-		left += conv.toProtoConversion + "("
-		right += ")"
+		if len(left) > 0 {
+			left += conv.toProtoConversion + "("
+			right = ")" + right
+		} else {
+			left += varName + " := " + conv.toProtoConversion + "("
+			right += ")"
+		}
+	}
+
+	if len(left) == 0 {
+		left += varName + " := "
 	}
+
 	return left + ident + right
 }
 
-func (g *serviceGenerator) renderToEnt(conv *converter, ident string) string {
+func (g *serviceGenerator) renderToEnt(conv *converter, varName, ident string) string {
```

Comment:
this method became a bit tough to parse mentally. i like the generated code, but maybe we can express this with an inline template for each use case?